src/views/Users/ViewUser/index.jsx

```diff
@@ -0,0 +1,470 @@
+            <Button
+              className={classes.fullWidth}
+              variant="outlined"
+              onClick={() => handleActionAdd(permission)}>
+              <PlusIcon />
+            </Button>
+          </Grid>
+        )}
+      </Grid>
+    ));
+
+  return (
+    <Dashboard title="Users">
+      {error && <ErrorPanel error={error} />}
+      {isLoading && <Spinner loading />}
+      {!isLoading && (
+        <Fragment>
+          <form autoComplete="off">
+            <Grid container spacing={2}>
+              <Grid item xs>
+                <TextField
+                  required
+                  disabled={!isNewUser}
+                  onChange={handleUsernameChange}
+                  fullWidth
+                  label="Username"
+                  value={username}
+                />
+              </Grid>
+            </Grid>
+            <br />
+            <br />
+            <br />
+            <Typography variant="h5">Roles</Typography>
+            {roles.map(renderRole)}
+            {additionalRoles.map(renderRole)}
+            <Grid item xs className={classes.addGrid}>
+              <Button
+                onClick={handleRoleAdd}
+                className={classes.fullWidth}
+                variant="outlined">
+                <PlusIcon />
+              </Button>
+            </Grid>
+            <br />
+            <br />
+            <br />
+            <Typography variant="h5">Permissions</Typography>
+            {permissions.map(renderPermission)}
+            {additionalPermissions.map(renderPermission)}
+            <Grid container spacing={2}>
+              <Grid item xs>
+                Name
+              </Grid>
+              <Grid item xs>
+                Product Restrictions
+              </Grid>
+              <Grid item xs>
+                Action Restrictions
+              </Grid>
+            </Grid>
+            <Grid item xs className={classes.addGrid}>
+              <Grid item xs={11}>
+                <Button
+                  onClick={handlePermissionAdd}
+                  className={classes.fullWidth}
+                  variant="outlined">
+                  <PlusIcon />
+                </Button>
+              </Grid>
+            </Grid>
+          </form>
+          <Tooltip title="Save User">
+            <Fab
+              disabled={saveAction.loading}
+              onClick={handleUserSave}
+              color="primary"
+              className={classes.fab}>
+              <ContentSaveIcon />
+            </Fab>
+          </Tooltip>
+          {!isNewUser && (
+            <SpeedDial ariaLabel="Secondary Actions">
+              <SpeedDialAction
+                disabled={saveAction.loading}
+                icon={<DeleteIcon />}
+                tooltipOpen
+                tooltipTitle="Delete Signoff"
+                onClick={handleUserDelete}
+              />
+            </SpeedDial>
+          )}
+        </Fragment>
+      )}
+    </Dashboard>
+  );
+}
+
```

Comment:
STR:

1. Navigate to the `balrogadmin` user view

![Screen Shot 2019-07-04 at 10 00 00 AM](https://user-images.githubusercontent.com/3766511/60672081-82bd8780-9e42-11e9-9ce3-d19a023e888b.png)

2. Click the "+" button to add a role
3. Modify the first text field with content "releng" by appending another character (e.g., `z`)

**Expected:** the text field with content "releng becomes "relengz".
**Actual:** the empty text field that was added is now "relengz" and the first text field did not change.